zerver/lib/streams.py

```diff
@@ -725,6 +726,10 @@ def list_to_streams(
                 raise JsonableError(_("Insufficient permission"))
             if not invite_only and not user_profile.can_create_public_streams():
                 raise JsonableError(_("Insufficient permission"))
+            if invite_only and is_default_stream:
+                raise JsonableError(
+                    _("A stream cannot be both a default stream for new users and private.")
+                )
```

Comment:
We also need a permissions check for whether the current user has permission to edit the set of default streams, both here and in the other code path. As written, anyone who can create a stream can make it one of the default streams for new users, which would be a significant security bug.

Reply:
I think the only check we need is whether the user is an admin, and I have added that check here. The other code paths already have this check. The error message is `Insufficient permission`, just like the other error messages here. Let me know if you want a more detailed error message.

Reply:
That works, thanks!